### PR TITLE
MQ: exchange: Introduce sock_recv for get_msgs from exchange

### PR DESCRIPTION
**Pull request by RanMaoyi**

### CODE REVIEW

File: src/mqtt/protocol/exchange/exchange_client.c [hunk 1]
```diff
@@ -249,6 +250,40 @@ exchange_sock_send(void *arg, nni_aio *aio)
...
+	nng_msg **list = NULL;
+
+	ret = exchange_client_get_msgs_by_key(s, key, count, &list);
+	if (ret != 0) {
+		log_error("exchange_client_get_msgs_by_key failed!\n");
+		nni_aio_finish_error(aio, NNG_EINVAL);
+		return;
+	}
+
+	nni_aio_set_msg(aio, list);
```
Comment (wanghaEMQ) [inline src/mqtt/protocol/exchange/exchange_client.c:279]: Will warning. Please fix it

File: src/mqtt/protocol/exchange/exchange_client.c [hunk 2]
```diff
@@ -354,30 +389,37 @@ int
...
-	exchange_node_t *ex_node = s->ex_node;
-	nni_mtx_lock(&ex_node->mtx);
-	/* Only one exchange with one ringBuffer now */
-	ret = ringBuffer_search_msgs_by_key(ex_node->ex->rbs[0], key, count, list);
-	if (ret != 0 || *list == NULL) {
-		log_error("ringBuffer_get_msgs_by_key failed!\n");
-		nni_mtx_unlock(&ex_node->mtx);
-		nni_mtx_unlock(&s->mtx);
-		return -1;
+	if (count == 1) {
```
Comment (wanghaEMQ) [inline src/mqtt/protocol/exchange/exchange_client.c:404]: Cool

**Review by wanghaEMQ — APPROVED** [review]

**Review by lee-emqx — APPROVED** [review]

**Review by OdyWayne — APPROVED** [review]

File: src/mqtt/protocol/exchange/exchange_client.c [hunk 3]
```diff
@@ -249,6 +250,40 @@ exchange_sock_send(void *arg, nni_aio *aio)
...
+
+	ret = exchange_client_get_msgs_by_key(s, key, count, &list);
+	if (ret != 0) {
+		log_error("exchange_client_get_msgs_by_key failed!\n");
+		nni_aio_finish_error(aio, NNG_EINVAL);
+		return;
+	}
+
+	nni_aio_set_msg(aio, list);
+	nni_aio_set_prov_data(aio, (void *)count);
```
Comment (wanghaEMQ) [inline src/mqtt/protocol/exchange/exchange_client.c:280]: nni_aio_set_prov_data(aio, (void *)(uintptr_t)count);

File: src/mqtt/protocol/exchange/exchange_client_test.c [hunk 4]
```diff
@@ -26,6 +26,23 @@ static inline void free_msg_list(nng_msg **msgList, nng_msg *msg, int *lenp, int
 	}
 }
 
+static inline void client_get_msgs(nng_socket sock, uint32_t key, int count, int *lenp, nng_msg ***msgList)
+{
+	nni_aio *aio = NULL;
+	NUTS_PASS(nng_aio_alloc(&aio, NULL, NULL));
+
+	nni_aio_set_prov_data(aio, (void *)key);
+	nni_aio_set_msg(aio, (void *)count);
```
Comment (wanghaEMQ) [inline src/mqtt/protocol/exchange/exchange_client_test.c:35]: will warning this two lines

**Review by JaylinYu — APPROVED** [review]

**Pull request merged**